subprojects/platform-base/src/main/java/org/gradle/language/base/internal/tasks/SimpleStaleClassCleaner.java

```diff
@@ -31,7 +31,7 @@ public SimpleStaleClassCleaner(TaskOutputsInternal taskOutputs) {
     public void execute() {
         String prefix = getDestinationDir().getAbsolutePath() + File.separator;
         for (File f : taskOutputs.getPreviousOutputFiles()) {
-            if (f.getAbsolutePath().startsWith(prefix)) {
+            if (f.getAbsolutePath().startsWith(prefix) && f.isFile()) {
```

Comment:
Shouldn't we be deleting directories here, too? Leftover empty directories make a difference in the resulting JARs etc.